draft-ietf-quic-transport.md

```diff
@@ -2419,44 +2447,53 @@ frame and has sent neither a FIN nor a RST_STREAM, it MUST send a RST_STREAM in
 response, bearing the offset of the last byte sent on this stream as the final
 offset.
 
-### Offset Increment
+### Data Limit Increments
 
-This document leaves when and how many bytes to advertise in a WINDOW_UPDATE to
-the implementation, but offers a few considerations.  WINDOW_UPDATE frames
-constitute overhead, and therefore, sending a WINDOW_UPDATE with small offset
-increments is undesirable.  At the same time, sending WINDOW_UPDATES with large
-offset increments requires the sender to commit to that amount of buffer.
-Implementations must find the correct tradeoff between these sides to determine
-how large an offset increment to send in a WINDOW_UPDATE.
+This document leaves when and how many bytes to advertise in a MAX_DATA or
+MAX_STREAM_DATA to implementations, but offers a few considerations.  These
+frames contribute to connection overhead.  Therefore frequently sending frames
+with small changes is undesirable.  At the same time, infrequent updates require
+larger increments to limits if blocking is to be avoided.  Thus, larger updates
+require a receiver to commit to larger resource commitments.  Thus there is a
+tradeoff between resource commitment and overhead when determining how large a
+limit is advertised.
 
-A receiver MAY use an autotuning mechanism to tune the size of the offset
-increment to advertise based on a roundtrip time estimate and the rate at which
-the receiving application consumes data, similar to common TCP implementations.
+A receiver MAY use an autotuning mechanism to tune the frequency and amount that
+it increases data limits based on a roundtrip time estimate and the rate at
+which the receiving application consumes data, similar to common TCP
+implementations.
 
 ### Stream Limit Increment
 
 As with flow control, this document leaves when and how many streams to make
-available to a peer via LIMIT_UPDATE to the implementation, but offers a few
-considerations. LIMIT_UPDATE frames constitute minimal overhead, while
-withholding LIMIT_UPDATEs prevents the peer from fully utilizing the transport.
+available to a peer via MAX_STREAM_ID to implementations, but offers a few
+considerations.  MAX_STREAM_ID frames constitute minimal overhead, while
+withholding MAX_STREAM_ID frames can prevent the peer from using the available
+parallelism.
+
+Implementations will likely want to increase the maximum stream ID as
+peer-initiated streams close.  Sending a MAX_STREAM_ID frame along with ACK
+frames ensures a regular increase in the stream limit without generating excess
```

Comment:
Not exactly.  Implies a particular strategy for increasing the maximum stream ID, which I want to avoid.  The point is that you might send any MAX_STREAM_ID frames you might want to send on the same schedule as ACK.  I'll try to improve it some.